tests/expr_and_series/mode_test.py

```diff
@@ -12,25 +14,23 @@
 }
 
 
-def test_mode_single_expr(
-    constructor_eager: ConstructorEager, request: pytest.FixtureRequest
-) -> None:
-    if "pyarrow_table" in str(constructor_eager):
-        # TODO(unassigned): reimplement mode for pyarrow
-        request.applymarker(pytest.mark.xfail)
+def test_mode_single_expr(constructor_eager: ConstructorEager) -> None:
     df = nw.from_native(constructor_eager(data))
     result = df.select(nw.col("a").mode()).sort("a")
     expected = {"a": [1, 2]}
     assert_equal_data(result, expected)
 
 
-def test_mode_series(
-    constructor_eager: ConstructorEager, request: pytest.FixtureRequest
-) -> None:
-    if "pyarrow_table" in str(constructor_eager):
-        # TODO(unassigned): reimplement mode for pyarrow
-        request.applymarker(pytest.mark.xfail)
```

Comment:
unexpected side-effect of this PR - fixing these two issues with pyarrow's mode 🥳